From: Tim <NightFlyer@users.noreply.github.com>
Date: Tue, 30 Apr 2019 17:41:08 -0700
Subject: Add XML attribute to tasklist (#145)

I'm not positive this is the best attribute to use, but at least
it doesn't lose the information as to whether the task was completed.
---
 extensions/tasklist.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/extensions/tasklist.c b/extensions/tasklist.c
index 509aca7..9caaadb 100644
--- a/extensions/tasklist.c
+++ b/extensions/tasklist.c
@@ -121,6 +121,15 @@ static void html_render(cmark_syntax_extension *extension,
   }
 }
 
+static const char *xml_attr(cmark_syntax_extension *extension,
+                            cmark_node *node) {
+  if ((int)node->as.opaque == CMARK_TASKLIST_CHECKED) {
+    return " completed=\"true\"";
+  } else {
+    return " completed=\"false\"";
+  }
+}
+
 cmark_syntax_extension *create_tasklist_extension(void) {
   cmark_syntax_extension *ext = cmark_syntax_extension_new("tasklist");
 
@@ -131,6 +140,7 @@ cmark_syntax_extension *create_tasklist_extension(void) {
   cmark_syntax_extension_set_commonmark_render_func(ext, commonmark_render);
   cmark_syntax_extension_set_plaintext_render_func(ext, commonmark_render);
   cmark_syntax_extension_set_html_render_func(ext, html_render);
+  cmark_syntax_extension_set_xml_attr_func(ext, xml_attr);
 
   return ext;
 }
